agent/src/trident.rs

```diff
@@ -682,6 +682,11 @@ impl Trident {
                 }
                 _ => (),
             }
+            // If `running_in_only_watch_k8s_mode` is set to true, no need to update components
```

Comment:
Thank you for your pr. 
api_watcher depends on the configuration issued by server, so you cannot continue it here.